src/rollup/types.d.ts

```diff
@@ -56,6 +56,7 @@ export interface SourceDescription {
 export interface ModuleJSON {
 	id: string;
 	dependencies: string[];
+	transformDependencies: string[];
```

Comment:
Even though these dependencies currently originate only from `transform` hooks, this is not really important to any later handling. Most notably, I could imagine the `load` hook declare dependencies as well (i.e. we load a .less file and the loader declares a bunch of dependencies right away).

Maybe we could call it something like `manualDependencies` or `additionalDependencies` or `nonModuleDependencies` or something like that.

Also, as stated, it might make sense to add this to the load hook as well (unless I miss something here).

Reply:
It is important for later handling due to the fact that they invalidate the module cache for their parent transform ids.

I think it's a simpler conceptual model to think in terms of transform dependencies invalidating that module. If there are dependencies of the build that don't invalidate any modules they are probably better emitted as assets anyway.

The load hook could support a dependencies array definitely, but it feels less clear whether that should invalidate the transform cache or not?

Another option might be to have `this.emitDependency` where the transform hook has its own context already. But my worry with that is that it isn't necessarily clear to trace what is invalidating what. The nice thing about the transform dependencies is that it is a very explicit invalidation chain.

Reply:
@guybedford you are right, `dependencies` solves [this issue](https://github.com/rollup/rollup/issues/1203#issuecomment-303883913) of rollup-plugin-pug